File: src/main/java/fr/arsene/alkoholspiele/model/documents/Game.java
package fr.arsene.alkoholspiele.model.documents;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.LinkedList;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Document
public class Game {

    @Id
    private String id;

    private String author;
    private String name;

    private List<Joke> jokes;

    public Game(String id, String author, String name) {
        this.id = id;
        this.author = author;
        this.name = name;
        this.jokes = new LinkedList<>();
    }
}

File: core/src/main/kotlin/alkoholspiele/core/entities/Game.kt
package alkoholspiele.core.entities

import java.util.UUID

class Game {
    val id: UUID
    val name: String;
    val author: String;
    val jokes: List<Joke>;

    constructor(name: String, author: String, jokes: List<Joke>) {
        this.name = name;
        this.author = author;
        id = UUID.randomUUID();
        this.jokes = jokes
    }

    constructor(id: UUID, name: String, author: String, jokes: List<Joke>) {
        this.name = name;
        this.author = author;
        this.id = id;
        this.jokes = jokes
    }


}